ocaml/lambda/translclass.ml

```diff
@@ -204,16 +206,22 @@ let rec build_object_init ~scopes cl_table obj params inh_init obj_init cl =
       (inh_init,
        let build params rem =
          let param = name_pattern "param" pat in
-         let param_layout =
-           Typeopt.layout pat.pat_env pat.pat_loc pat.pat_type
+         let arg_sort = Sort.for_class_arg in
```

Comment:
In this PR we want to do the minimal thing of only using front-end's sort/layout calculations in the places needed for the uses of unboxed floats we're ready to support.  When we want to allow class arguments of other sorts, we'll delete `Sort.for_class_arg` and make the change you're proposing then.

---------------------------

As an aside:

Patterns don't really carry a sort.  They do carry a type, and we can calculate a sort from the type if we need to, but it's a goal of ours to avoid ever performing such a calculation during the translation to lambda (both because it's potentially expensive, and because we should be "done" with that kind of typechecker code once we've computed the typed tree).

When we're ready to delete `Sort.for_class_arg`, I think the right thing will be to add the sort of the argument to `Tcl_fun`, much like we've already done in `Texp_function`.  This way, we'll calculate the sort during typechecking and not need to extract it from the pattern.